Document TRL and DSOS

mlenkeit opened the pull request:
This PR aims at fixing #832.

mlenkeit committed:
docs: add DSOS diagrams

Ein-Tim commented:
What is blocking this PR from being merged?

MikeMcC399 commented:
@Ein-Tim 
Currently the PR is a collection of images which are not embedded anywhere in a document, so it is incomplete. Perhaps @mlenkeit can let us know his plans?

MikeMcC399 commented:
@mlenkeit 
Are you going to continue with this PR sometime?

mlenkeit commented:
> @mlenkeit Are you going to continue with this PR sometime?

yes.

mlenkeit committed:
Merge branch 'main' into trl-and-dsos

mlenkeit committed:
docs(solar): fix ENF name in diagram

use Exposure Notification Framework instead of Event Notification Framework

mlenkeit committed:
docs(solar): describe DSOS and TRL mapping

mlenkeit committed:
docs(solar): drop Attenuation Bucket chapter

with ENF v2 and exposure windows, the attenuation buckets are nothing
extraordinary anymore that would justify a separate chapter. It's just
another configuration parameter during the risk calculation and
described as such.

mlenkeit commented:
I have reworked the chapters on the risk calculation to make the role of TRL and DSOS and the respective mapping a bit clearer.

@MikeMcC399 I would appreciate your feedback if you find some time.

mlenkeit committed:
chore: allow details and summary tags in md

MikeMcC399 commented on solution_architecture.md:
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
+
+    </details>
+
+2. **No symptoms** - the user indicates that he/she has symptoms
```
"the user indicates that he/she has symptoms" =>

"the user indicates that he/she has no symptoms" ?

mlenkeit replied:
Thanks, good catch (as usual) 👍 Fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/b1505f1cb5a031acd2586d89ec567282a2649c23.

MikeMcC399 reviewed (changes requested):
Please check the comment about no symptoms.

MikeMcC399 commented:
@mlenkeit 

> I have reworked the chapters on the risk calculation to make the role of TRL and DSOS and the respective mapping a bit clearer.

Many thanks!

> @MikeMcC399 I would appreciate your feedback. I'll leave the PR open for a few days.

I only did a brief sanity check on the changes and I have just one minor comment to take a look at. Otherwise it looks good to merge.

mlenkeit committed:
docs(solar): fix typo in DSOS explanation

MikeMcC399 reviewed (approved):
LGTM!

Ein-Tim commented on solution_architecture.md:
```diff
@@ -225,24 +229,24 @@ For Apple devices an OS version of at least 12.5 (for older devices) or 13.7 is
 
+### ENF Usage
+
+The Exposure Notification Framework (ENF) encapsulates handling of the keys, including all cryptographic operations on them, and broadcasting and scanning for Bluetooth advertisement packages for the app (see *Figure 11* and *Figure 12*). The app itself does not have access to the collected exposures, i.e. the Rolling Proximity Identifiers, and neither is it informed if a new one has been collect ...
```
Suggestion: Add the information that the app also is not informed if these keys are deleted. This is quite an important info and answers questions like: "Why doesn't the app change if I delete the keys via OS settings?"

mlenkeit replied:
Fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/0e19cdfc1325f4af10322efb5237cc471cef2750.

Ein-Tim commented on solution_architecture.md:
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
+### Risk Calculation
 
-While in the first version of the Exposure Notification Framework, Apple and Google allowed to define multiple thresholds for the attenuation, this functionality can now be implemented within the application through the data from the exposure windows.
+The Exposure Windows form the foundation of the risk calculation in the app. The result is an easy-to-understand risk level that is displayed to the user: low risk (i.e. _green_) or high risk _red_. The details of the risk calculation, e.g. exposure windows or intermediate results, are not visible to the user. Further information regarding the individual exposure events (such as the matched Rolli ...
```
Add brackets to the word "red".

```suggestion
The Exposure Windows form the foundation of the risk calculation in the app. The result is an easy-to-understand risk level that is displayed to the user: low risk (i.e. _green_) or high risk (_red_). The details of the risk calculation, e.g. exposure windows or intermediate results, are not visible to the user. Further information regarding the individual exposure events (such as the matched Rolling Proximity Identifier, the Temporary Exposure Key or the exact time) remains within the secure storage of the framework and cannot be retrieved by the application.
```

Ein-Tim commented on solution_architecture.md:
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
+
+    </details>
+
+2. **No symptoms** - the user indicates that he/she has no symptoms
```
Use they to be gender neutral.

```suggestion
2. **No symptoms** - the user indicates that they have no symptoms
```

mlenkeit replied:
@Ein-Tim I like the proposal of dropping _he/she_ but I'm not sure if _they_ is such a good replacement. How about just dropping the relative clause and rephrase it to _the user has no symptoms_?

Ein-Tim replied:
Sounds good!

mlenkeit replied:
fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/19341c3fef9edf172d14c7773f933263b71ddefc

Ein-Tim commented on solution_architecture.md:
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
+
+    </details>
+
+3. **No date information** - the user indicates that he/she has symptoms but declines to provide any date information
```
```suggestion
3. **No date information** - the user indicates that they have symptoms but decline to provide any date information
```

Ein-Tim commented on solution_architecture.md:
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
+
+    </details>
+
+4. **Date range** - the user indicates that he/she has symptoms and provides a date range (e.g. "within the last week", "1 to 2 weeks ago", "more than 2 weeks ago")
```
```suggestion
4. **Date range** - the user indicates that they have symptoms and provide a date range (e.g. "within the last week", "1 to 2 weeks ago", "more than 2 weeks ago")
```

Ein-Tim commented on solution_architecture.md:
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
+
+    </details>
+
+5. **Specific date** - the user indicates that he/she has symptoms and provides a specific date
```
```suggestion
5. **Specific date** - the user indicates that they have symptoms and provide a specific date
```

Ein-Tim reviewed (commented):
Some suggestions, e.g. to use "they" instead of "he/she". but in general: Looks good to me!

mlenkeit committed:
docs(solar): mention card color with risk level

Co-authored-by: Tim <67682506+Ein-Tim@users.noreply.github.com>

mlenkeit committed:
docs(solar): rephrase app access to RPIs

mlenkeit committed:
Merge branch 'trl-and-dsos' of https://github.com/corona-warn-app/cwa-documentation into trl-and-dsos

mlenkeit committed:
docs(solar): drop pronouns he/she

Ein-Tim reviewed (approved):
LGTM!
